python/tskit/tables.py

```diff
@@ -585,6 +586,42 @@ def _update_metadata_schema_cache_from_ll(self) -> None:
             self.ll_table.metadata_schema
         )
 
+    def metadata_vector(self, key, *, dtype=None, default_value=None):
```

Comment:
Just an idea but you could make `key` optional and return the whole metadata object if it is not specified?

Reply:
Hm - nice idea, but that might encourage bad habits? Currently, metadata has to be a dictionary (not eg a single number), so I'm not sure about this. We could always extend it later to cover this case without breaking this API.

Reply:
I agree with Peter here